apps/studio/pages/project/[ref]/functions/new.tsx

```diff
@@ -0,0 +1,286 @@
+                <CommandList_Shadcn_>
+                  <CommandEmpty_Shadcn_>No templates found.</CommandEmpty_Shadcn_>
+                  <CommandGroup_Shadcn_>
+                    {EDGE_FUNCTION_TEMPLATES.map((template) => (
+                      <CommandItem_Shadcn_
+                        key={template.value}
+                        value={template.value}
+                        onSelect={onSelectTemplate}
+                        onMouseEnter={() => handleTemplateMouseEnter(template.content)}
+                        onMouseLeave={handleTemplateMouseLeave}
+                        className="cursor-pointer"
+                      >
+                        <div className="flex flex-col gap-1">
+                          <div className="flex items-center">
+                            <Check
+                              className={cn(
+                                'mr-2 h-4 w-4',
+                                files.some((f) => f.selected && f.content === template.content)
+                                  ? 'opacity-100'
+                                  : 'opacity-0'
+                              )}
+                            />
+                            <span className="text-foreground">{template.name}</span>
+                          </div>
+                          <span className="text-xs text-foreground-light pl-6">
+                            {template.description}
+                          </span>
+                        </div>
+                      </CommandItem_Shadcn_>
+                    ))}
+                  </CommandGroup_Shadcn_>
+                </CommandList_Shadcn_>
+              </Command_Shadcn_>
+            </PopoverContent_Shadcn_>
+          </Popover_Shadcn_>
+          <Button
+            size="tiny"
+            type="default"
+            onClick={handleChat}
+            icon={<AiIconAnimation size={16} />}
+          >
+            Chat
+          </Button>
+        </>
+      }
+    >
+      <FileExplorerAndEditor
+        files={files}
+        onFilesChange={setFiles}
+        aiEndpoint={`${BASE_PATH}/api/ai/edge-function/complete`}
+        aiMetadata={{
+          projectRef: project?.ref,
+          connectionString: project?.connectionString,
+          includeSchemaMetadata,
+        }}
+      />
+
+      <div className="flex items-center bg-background-muted justify-end p-4 border-t bg-surface-100">
+        <Button
+          loading={isDeploying}
+          size="medium"
+          disabled={!functionName || files.length === 0}
+          onClick={onDeploy}
+          iconRight={
+            isDeploying ? (
+              <Loader2 className="animate-spin" size={10} strokeWidth={1.5} />
+            ) : (
+              <div className="flex items-center space-x-1">
+                <CornerDownLeft size={10} strokeWidth={1.5} />
+              </div>
+            )
+          }
+        >
+          Deploy function
```

Comment:
shall we hook up a keyboard shortcut? the enter icon makes it seem like there's a keyboard shortcut to it
let's make it consistent across too like SQL Editor, Assistant, etc -> add the Cmd icon

![image](https://github.com/user-attachments/assets/5d654237-7e71-4555-9da1-bdd224504bd0)